plugin/core/logging.py

```diff
@@ -39,3 +39,33 @@ def exception_log(message: str, ex: Exception) -> None:
 def printf(*args: Any, prefix: str = 'LSP') -> None:
     """Print args to the console, prefixed by the plugin name."""
     print(prefix + ":", *args)
+
+
+def notify(window: Optional[sublime.Window], message: str, status: str = 'LSP: see console log…') -> None:
+    """Pick either of the 2 ways to show a user notification message:
+      - via a detailed console message and a short status message
+      - via a blocking modal dialog"""
+    from .settings import userprefs
+    if userprefs().suppress_error_dialogs:
+        if window:
+            window.status_message(status)  # print short message to statusbar
+        else:
+            sublime.status_message(status)
+        print(message)  # print full message to console log
+    else:
+        sublime.message_dialog(message)
+
+
+def notify_error(window: Optional[sublime.Window], message: str, status: str = '❗LSP: see console log…') -> None:
```

Comment:
I think we don't need 2 separate functions. Current use of `sublime.mesage_dialog` and `sublime.error_message` looks pretty arbitrary. I think all of them are errors, so I'd use only `sublime.error_message`. If the severity is important it could be another parameter `severety`, but I don't think that's necessary. `notify_error` sounds like the appropriate function name, so I would remove the other function.